From f18d0189d6069326259e54f58ed0a8d693bd5237 Mon Sep 17 00:00:00 2001
From: Mika Laitio <mika.laitio@amd.com>
Date: Tue, 14 Oct 2025 15:10:44 -0700
Subject: [PATCH] rocfft-test and hipfft-test linking fix on windows

Link rocfft-test and hipfft-test against
fftw3.lib and fftw3f.lib instead of fftw3.dll and fftw3f.dll
when linking against the fftw3-libraries build by therock.

Fixes following errors:

[rocFFT] lld-link: error:
C:/TheRock/build/third-party/fftw3/build_fftw3/dist/bin/fftw3.dll:
bad file type. Did you specify a DLL instead of an import libraryi

[rocFFT] lld-link: error:
C:/TheRock/build/third-party/fftw3/build_fftw3f/dist/bin/fftw3f.dll:
bad file type. Did you specify a DLL instead of an import library?

Signed-off-by: Mika Laitio <mika.laitio@amd.com>
---
 projects/hipfft/clients/tests/CMakeLists.txt | 9 +++++++--
 projects/rocfft/clients/tests/CMakeLists.txt | 9 +++++++--
 2 files changed, 14 insertions(+), 4 deletions(-)

diff --git a/projects/hipfft/clients/tests/CMakeLists.txt b/projects/hipfft/clients/tests/CMakeLists.txt
index b37e817ad3..a424c68b87 100644
--- a/projects/hipfft/clients/tests/CMakeLists.txt
+++ b/projects/hipfft/clients/tests/CMakeLists.txt
@@ -70,8 +70,13 @@ else()
   find_package(FFTW3f CONFIG QUIET )
   if( FFTW3_FOUND AND FFTW3f_FOUND )
     message(STATUS "FFTW3 found by using the FFTW3Config.cmake")
-    get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_LOCATION_RELEASE)
-    get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_LOCATION_RELEASE)
+    if ( NOT WIN32 )
+      get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_LOCATION_RELEASE)
+      get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_LOCATION_RELEASE)
+    else()
+      get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_IMPLIB_RELEASE)
+      get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_IMPLIB_RELEASE)
+    endif()
 
     # Look for omp (preferred) or thread libraries. These are not a
     # hard requirement, but are nice to have to make FFTW run faster.
diff --git a/projects/rocfft/clients/tests/CMakeLists.txt b/projects/rocfft/clients/tests/CMakeLists.txt
index 7ebda11f07..8684c5cce2 100644
--- a/projects/rocfft/clients/tests/CMakeLists.txt
+++ b/projects/rocfft/clients/tests/CMakeLists.txt
@@ -128,8 +128,13 @@ if( NOT BUILD_FFTW )
     find_package(FFTW3f CONFIG QUIET )
     if( FFTW3_FOUND AND FFTW3f_FOUND )
       message(STATUS "FFTW3 found by using the FFTW3Config.cmake")
-      get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_LOCATION_RELEASE)
-      get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_LOCATION_RELEASE)
+      if ( NOT WIN32 )
+        get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_LOCATION_RELEASE)
+        get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_LOCATION_RELEASE)
+      else()
+        get_target_property(fftw3_libs FFTW3::fftw3 IMPORTED_IMPLIB_RELEASE)
+        get_target_property(fftw3f_libs FFTW3::fftw3f IMPORTED_IMPLIB_RELEASE)
+      endif()
 
       # Look for omp (preferred) or thread libraries. These are not a
       # hard requirement, but are nice to have to make FFTW run faster.
-- 
2.43.0

